packages/docusaurus-utils-validation/src/validationSchemas.ts

```diff
@@ -6,6 +6,9 @@
  */
 import * as Joi from '@hapi/joi';
 
+// it is from Joi core with support of more than one hash
+const URLRegex = /^(?=.)(?!https?:\/$)(?:[a-zA-Z][a-zA-Z\d+-.]*:(?:(?:\/\/(?:[\w-.~%\dA-Fa-f!$&'()*+,;=:]*@)?(?:\[(?:(?:(?:[\dA-Fa-f]{1,4}:){6}(?:[\dA-Fa-f]{1,4}:[\dA-Fa-f]{1,4}|(?:(?:0{0,2}\d|0?[1-9]\d|1\d\d|2[0-4]\d|25[0-5])\.){3}(?:0{0,2}\d|0?[1-9]\d|1\d\d|2[0-4]\d|25[0-5]))|::(?:[\dA-Fa-f]{1,4}:){5}(?:[\dA-Fa-f]{1,4}:[\dA-Fa-f]{1,4}|(?:(?:0{0,2}\d|0?[1-9]\d|1\d\d|2[0-4]\d|25[0-5])\.){3}(?:0{0,2}\d|0?[1-9]\d|1\d\d|2[0-4]\d|25[0-5]))|(?:[\dA-Fa-f]{1,4})?::(?:[\dA-Fa-f]{1,4}:){4}(?:[\dA-Fa-f]{1,4}:[\dA-Fa-f]{1,4}|(?:(?:0{0,2}\d|0?[1-9]\d|1\d\d|2[0-4]\d|25[0-5])\.){3}(?:0{0,2}\d|0?[1-9]\d|1\d\d|2[0-4]\d|25[0-5]))|(?:(?:[\dA-Fa-f]{1,4}:){0,1}[\dA-Fa-f]{1,4})?::(?:[\dA-Fa-f]{1,4}:){3}(?:[\dA-Fa-f]{1,4}:[\dA-Fa-f]{1,4}|(?:(?:0{0,2}\d|0?[1-9]\d|1\d\d|2[0-4]\d|25[0-5])\.){3}(?:0{0,2}\d|0?[1-9]\d|1\d\d|2[0-4]\d|25[0-5]))|(?:(?:[\dA-Fa-f]{1,4}:){0,2}[\dA-Fa-f]{1,4})?::(?:[\dA-Fa-f]{1,4}:){2}(?:[\dA-Fa-f]{1,4}:[\dA-Fa-f]{1,4}|(?:(?:0{0,2}\d|0?[1-9]\d|1\d\d|2[0-4]\d|25[0-5])\.){3}(?:0{0,2}\d|0?[1-9]\d|1\d\d|2[0-4]\d|25[0-5]))|(?:(?:[\dA-Fa-f]{1,4}:){0,3}[\dA-Fa-f]{1,4})?::[\dA-Fa-f]{1,4}:(?:[\dA-Fa-f]{1,4}:[\dA-Fa-f]{1,4}|(?:(?:0{0,2}\d|0?[1-9]\d|1\d\d|2[0-4]\d|25[0-5])\.){3}(?:0{0,2}\d|0?[1-9]\d|1\d\d|2[0-4]\d|25[0-5]))|(?:(?:[\dA-Fa-f]{1,4}:){0,4}[\dA-Fa-f]{1,4})?::(?:[\dA-Fa-f]{1,4}:[\dA-Fa-f]{1,4}|(?:(?:0{0,2}\d|0?[1-9]\d|1\d\d|2[0-4]\d|25[0-5])\.){3}(?:0{0,2}\d|0?[1-9]\d|1\d\d|2[0-4]\d|25[0-5]))|(?:(?:[\dA-Fa-f]{1,4}:){0,5}[\dA-Fa-f]{1,4})?::[\dA-Fa-f]{1,4}|(?:(?:[\dA-Fa-f]{1,4}:){0,6}[\dA-Fa-f]{1,4})?::)|v[\dA-Fa-f]+\.[\w-.~!$&'()*+,;=:]+)\]|(?:(?:0{0,2}\d|0?[1-9]\d|1\d\d|2[0-4]\d|25[0-5])\.){3}(?:0{0,2}\d|0?[1-9]\d|1\d\d|2[0-4]\d|25[0-5])|[\w-.~%\dA-Fa-f!$&'()*+,;=]{1,255})(?::\d*)?(?:\/[\w-.~%\dA-Fa-f!$&'()*+,;=:@]*)*)|\/(?:[\w-.~%\dA-Fa-f!$&'()*+,;=:@]+(?:\/[\w-.~%\dA-Fa-f!$&'()*+,;=:@]*)*)?|[\w-.~%\dA-Fa-f!$&()*+,;=:@]+(?:\/[\w-.~%\dA-Fa-f!$&'()*+,;=:@]*)*|(?:\/\/\/[\w-.~%\dA-Fa-f!$&'()*+,;=:@]*(?:\/[\w-.~%\dA-Fa-f!$&'()*+,;=:@]*)*)))(?:\?[\w-.~%\dA-Fa-f!$&'()*+,;=:@/?]*(?=#|$))?(?:#[\w-.~%\dA-Fa-f!$&'()*+,;=:@/?#]*)?$/;
```

Comment:
I'm not totally confortable introducing such a complex regex into our codebase.

If we do, we should have a way to clearly identity how it diverge from the original Joi code it was copied from.


I'd prefer a way simpler alternative such as:

```
const CustomURLSchema = Joi.string().custom((value, helpers) => {
  try { 
     new URL(value);
     return value;
   } catch(e) {
	 helpers.error('any.invalid');
   }
}, 'is valid URL');
```

It is maybe not necessarily better, but at least we understand what's happening easily, and it's not a pain to fix bugs by editing the weird regex 🤪

Reply:
Fixed